src/Microsoft.DocAsCode.Build.Engine/Incrementals/DependencyItem.cs

```diff
@@ -5,35 +5,46 @@ namespace Microsoft.DocAsCode.Build.Engine.Incrementals
 {
     using System;
 
-    using Microsoft.DocAsCode.Common;
-
     using Newtonsoft.Json;
 
     public sealed class DependencyItem : IEquatable<DependencyItem>
     {
-        private static readonly StringComparer Comparer = FilePathComparer.OSPlatformSensitiveStringComparer;
-
         [JsonProperty("from")]
-        public string From { get; }
+        public DependencyItemSourceInfo From { get; private set; }
 
         [JsonProperty("to")]
-        public string To { get; }
+        public DependencyItemSourceInfo To { get; private set; }
 
         [JsonProperty("reportedBy")]
-        public string ReportedBy { get; }
+        public DependencyItemSourceInfo ReportedBy { get; private set; }
 
         [JsonProperty("type")]
         public string Type { get; }
 
         [JsonConstructor]
-        public DependencyItem(string from, string to, string reportedBy, string type)
+        public DependencyItem(DependencyItemSourceInfo from, DependencyItemSourceInfo to, DependencyItemSourceInfo reportedBy, string type)
         {
             From = from;
             To = to;
             ReportedBy = reportedBy;
             Type = type;
         }
 
+        public void UpdateFrom(DependencyItemSourceInfo f)
```

Comment:
Or create new DependencyItem instance?

---
In reply to: [102122717](https://github.com/dotnet/docfx/pull/1323#discussion_r102122717) [](ancestors = 102122717)